src/Ditto.Kinesis/ReplicatingConsumer.cs

```
@@ -0,0 +1,128 @@
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            StreamName = streamName ?? throw new ArgumentNullException(nameof(streamName));
+            GroupName = groupName ?? throw new ArgumentNullException(nameof(groupName));
+        }
+
+        public string StreamName { get; }
+        public string GroupName { get; }
+        public bool CanConsume(string eventType) => true;
+
+        public async Task ConsumeAsync(string eventType, ResolvedEvent resolvedEvent)
+        {
+            if (string.IsNullOrWhiteSpace(eventType)) throw new ArgumentException("Event type required", nameof(eventType));
+
+            if (_dittoSettings.ReadOnly)
+            {
+                _logger.Debug("Received {EventType} #{EventNumber} from {StreamName} (Original Event: #{OriginalEventNumber})",
+                    resolvedEvent.Event.EventType,
+                    resolvedEvent.Event.EventNumber,
+                    resolvedEvent.Event.EventStreamId,
+                    resolvedEvent.OriginalEventNumber
+                );
+
+                return;
+            }
+
+            var request = new PutRecordRequest
+            {
+                StreamName = _kinesisSettings.StreamName,
+                PartitionKey = resolvedEvent.Event.EventStreamId,
+                Data = CreateDataStream(resolvedEvent)
+            };
+
+            using (_logger.OperationAt(LogEventLevel.Debug).Time("Replicating {EventType} #{EventNumber} from {StreamName} (Original Event: #{OriginalEventNumber}) to Kinesis",
+                resolvedEvent.Event.EventType,
+                resolvedEvent.Event.EventNumber,
+                resolvedEvent.Event.EventStreamId,
+                resolvedEvent.OriginalEventNumber))
+            using (DittoMetrics.IODuration.WithIOLabels("kinesis", _kinesisSettings.StreamName, "put_record").NewTimer())
+            {
+                await _kinesis.PutRecordAsync(request);
```

We ideally want to batch write multiple events.

---

Is there any retry strategies here? You'll see two exceptions in normal operation, `ProvisionedThroughputExceededException` and occasionally `AmazonKinesisException` when something goes wrong in AWS systems. Typically I catch all exceptions around put and just do a simple Polly exponential backoff with jitter, could be useful in the case of having many Dittos.

---

As discussed, we can use the Standard retry mode by setting an environment variable in AWS